crates/rust-analyzer/src/reload.rs

```diff
@@ -474,7 +474,10 @@ impl GlobalState {
                 .all(|(l, r)| l.eq_ignore_build_data(r));
 
         if same_workspaces {
-            let (workspaces, build_scripts) = self.fetch_build_data_queue.last_op_result();
+            let Some((workspaces, build_scripts)) = self.fetch_build_data_queue.last_op_result()
+            else {
+                return;
+            };
```

Comment:
This is wrong, this means if we reload the workspace before building build scripts we'll stop updating things entirely. Previously we only early returned inthis case if `!Arc::ptr_eq(workspaces, &self.workspaces)` (and even then we mightve done rebuilt the crate graph still)

Reply:
oh, yeah. I see what you mean. fixed.